src/main/java/core/basesyntax/service/impl/FileReaderImpl.java

```diff
@@ -0,0 +1,31 @@
+package core.basesyntax.service.impl;
+
+import core.basesyntax.service.Reader;
+import java.io.BufferedReader;
+import java.io.FileReader;
+import java.io.IOException;
+import java.util.ArrayList;
+import java.util.List;
+
+public class FileReaderImpl implements Reader {
+    @Override
+    public List<String> readFile(String path) {
+        List<String> rows = new ArrayList<>();
+        BufferedReader reader;
```

Comment:
Consider using a try-with-resources statement to automatically close the `BufferedReader`. This will ensure that the resource is closed properly even if an exception occurs, improving resource management.